src/dmd/escape.d

```diff
@@ -2430,3 +2447,38 @@ private bool setUnsafeDIP1000(Scope* sc, bool gag, Loc loc, const(char)* msg, Ro
         return false;
     }
 }
+
+/***************************************
+ * Check that taking the address of `v` is `@safe` because of transitive scope.
+ *
+ * It's not possible to take the address of a scope variable, because `scope` only applies
+ * to the top level indirection.
+ *
+ * Params:
+ *     v = variable that a reference is created
+ *     e = expression that takes the referene
+ *     sc = used to obtain function / deprecated status
+ *     gag = don't print errors
+ * Returns:
+ *     true if taking the address of `v` is problematic because of the lack of transitive `scope`
+ */
+private bool checkTransitiveScope(VarDeclaration v, Expression e, Scope* sc, bool gag)
+{
+    if (!(v.storage_class & STC.temp) && v.isScope())
+    {
```

Comment:
Please rewrite this function to use early returns. That will diminish the nesting level and will make it easier to read the code:

```d
if (!e.type)
    return false;

if (v.storage_class & STC.temp || !v.isScope)
    return false;

Type t = e.type.baseElemOf();
....

```

Reply:
Done